descriptors/descriptors.go

```diff
@@ -35,6 +41,33 @@ func (d *Descriptor) String() string {
 	return d.mod.descriptorString(d.ptr)
 }
 
-func (d *Descriptor) AddressAt(network Network, multipathIndex uint32, derivationIndex uint32) (string, error) {
-	return d.mod.descriptorAddressAt(d.ptr, network, multipathIndex, derivationIndex)
+func (d *Descriptor) AddressAt(network Network, multipathIndex uint32,
+	derivationIndex uint32) (string, error) {
+
+	return d.mod.descriptorAddressAt(
+		d.ptr, network, multipathIndex, derivationIndex,
+	)
+}
+
+type Miniscript struct {
+	ParsedTypes string
+	OpCodes     uint64
+}
+
+func ParseMiniscript(script string) (*Miniscript, error) {
```

Comment:
What is exactly the purpose of these two functions that cannot be accomplished with the Descriptor struct? Can't you call NewDescriptor() and to add methods to get the pkScript.

This one specifically only returns types and the op code count, but for what purpose? 

Also this seems to hardcode the SegWit context, which seems arbitrary.